typescript/widgets/src/walletIntegrations/multiProtocol.tsx

```diff
@@ -127,14 +128,55 @@ export function getAccountAddressForChain(
   if (!chainName || !accounts) return undefined;
   const protocol = multiProvider.getProtocol(chainName);
   const account = accounts[protocol];
-  if (protocol === ProtocolType.Cosmos) {
+  if (
+    protocol === ProtocolType.Cosmos ||
+    protocol === ProtocolType.CosmosNative
+  ) {
     return account?.addresses.find((a) => a.chainName === chainName)?.address;
   } else {
     // Use first because only cosmos has the notion of per-chain addresses
     return account?.addresses[0]?.address;
   }
 }
 
+export function getAddressFromAccountAndChain(
+  account?: AccountInfo,
```

Comment:
When would these values be undefined?

Reply:
During initialization of the page when the account is not "ready" yet it can be undefined